### DISCUSSION TITLE
Ffmpeg scale arguments override

### DISCUSSION THREAD

**Discussion by drejkus** [blakeblackshear/frigate#5735] [General]
Hello, 
is it possible to somehow override (or leave out) scale arguments to ffmpeg (detect)? 
Using preset is not an option for me (none fits). (Maybe if there was a preset with empty scaling?)

As for me, leaving out scaling can have significant performance impact. 
(I'm trying to utilize hwaccel on rk3399, but sw scaling is incompatible with drm_prime format)

Thank you.

**Comment by blakeblackshear** [comment]
You see a performance benefit even when it's scaling to the native resolution of the camera?

**Reply by drejkus** [reply]
Well, I investigated that if I drop `-pix_fmt` and `-s`, ffmpeg can go decode fully in harware (`-hwaccel drm -hwaccel_output_format drm_prime`). If there is `-s` (even with the same resolution as input), ffmpeg fails with:

`Impossible to convert between the formats supported by the filter 'Parsed_null_0' and the filter 'auto_scale_0'`

Another issue would be that leaving out `pix_fmt` results in wrong pixel format (probably `nv12`), which of course produces weirdly colored image with artifacts, but recordings seem to be fine so I could live with that.

**Reply by blakeblackshear** [reply]
For now, Frigate requires the pixel format to be yuv420. We may support nv12 in the future. Detection wont really work with all the artifacts. This doesn't impact recordings because they are just copied to disk directly. These arguments are not used at all for recordings. There isn't currently a way to remove the resize operation without modifying the source code.

**Reply by drejkus** [reply]
Oh, I see. So for now I'll leave it to CPU. Thanks for clarification.
